assignment/scripts/function-practice.js

```diff
@@ -50,24 +55,39 @@ console.log( 'isPositive - should say false', isPositive(-3) );
 // 6. Function to return the _last_ item in an array. If the 
 //    array is empty, return `undefined`.
 function getLast( array ) {
-
+  return array[array.length -1];
 }
 
+console.log('The last item in the array is', getLast([1,3,5]));
+console.log('The array is:', getLast([]))
+
 // 7. Function to find a value in an array. Return true if the 
 //    value is found and false otherwise. Use a loop;
 //    DO NOT use Array.includes, Array.indexOf, or Array.find 
 function find( value, array ){
-  
+  for (let object of array){
+    if (object == value){
+      return true;
+    }
+  } return false;
 }
 
+console.log(find(7, [1,3,5,7]))
+console.log(find(7, [2,4,6,8]))
+
 // ----------------------
 // Stretch Goals
 // ----------------------
 // 8. Function to check if a letter is the first letter in a 
 //    string. Return true if it is, and false otherwise
 function isFirstLetter(letter, string) {
-
+  for (let specificLetter of string){
+    if (specificLetter == letter){
+      return true;
```

Comment:
Not quite. This will return true if the first letter matches the passed letter. It will also return true if the _second_ letter matches the passed letter. And the third, and the fourth, and so on. You wrote a function which checks for a letter _anywhere_ in the string.

Reply:
On second look, I see now that your `return false` is _inside_ the loop. That means this _will_ work, because you will always return _something_ on the first loop iteration, but it seems like you almost got a working implementation by accident here.

This would have been far more understandable:

```js
function isFirstLetter(letter, string) {
  return string[0] === letter;
}
```